client/packages/programs/src/JsonForms/components/PeriodSearch.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState } from 'react';
+import { rankWith, ControlProps, uiTypeIs } from '@jsonforms/core';
+import { z } from 'zod';
+import {
+  Autocomplete,
+  DetailInputWithLabelRow,
+  extractProperty,
+} from '@openmsupply-client/common';
+import {
+  DefaultFormRowSx,
+  FORM_LABEL_WIDTH,
+  useZodOptionsValidation,
+} from '../common';
+import { useJsonForms, withJsonFormsControlProps } from '@jsonforms/react';
+import { usePeriodList } from '../../api/hooks/usePeriodList';
+import { PeriodFragment } from '@openmsupply-client/requisitions';
+
+export const periodSearchTester = rankWith(10, uiTypeIs('PeriodSearch'));
+
+const Options = z.object({
+  findByProgram: z.boolean().optional(),
+});
+
+const UIComponent = (props: ControlProps) => {
+  const { handleChange, label, path, uischema } = props;
+  const [period, setPeriod] = useState<PeriodFragment | null>(null);
+  const { options } = useZodOptionsValidation(Options, uischema.options);
+  const { core } = useJsonForms();
+  const programId = options?.findByProgram
+    ? extractProperty(core?.data, 'programId')
+    : null;
+  const { data, isLoading } = usePeriodList(
+    programId,
+    options?.findByProgram ? !!programId : false
+  );
+
+  const onChange = async (period: PeriodFragment) => {
+    setPeriod(period);
+    handleChange(path, period.id);
```

Comment:
Will have to refactor this in other report issue to also return the start/end date to correctly filter out stocktake/prescriptions by start/end time